### PR TITLE
Docs for run templates

### PR DESCRIPTION
**Pull request by bcdurak**
## Describe changes
Docs pages for run templates. [Gitbook Space](https://zenml-io.gitbook.io/barisky-docs/how-to/trigger-pipelines) to check the changes

## Pre-requisites
Please ensure you have done the following:
- [X] I have read the **CONTRIBUTING.md** document.
- [X] If my change requires a change to docs, I have updated the documentation accordingly.
- [ ] I have added tests to cover my changes.
- [X] I have based my new branch on `develop` and the open PR is targeting `develop`. If your branch wasn't based on develop read [Contribution guide on rebasing branch to develop](https://github.com/zenml-io/zenml/blob/main/CONTRIBUTING.md#-pull-requests-rebase-your-branch-on-develop).
- [X] If my changes require changes to the dashboard, these changes are communicated/requested.

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [X] Other (add details above): Docs

### CODE REVIEW

**Review by htahir1 — CHANGES_REQUESTED** [review]
This is great! However i just noticed that this section of the docs https://docs.zenml.io/how-to/trigger-pipelines somehow seems disconnected from the one you edited and perhaps we should merge it? WDYT?

File: docs/book/how-to/create-and-run-templates/run-a-template.md [hunk 1]
```diff
@@ -1,30 +1,51 @@
...
+
+In order to run a template from the dashboard:
+
+- You can either click `Run a Pipeline` on the main `Pipelines` page, or
+- You can go to a specific template page and click on `Run Template`.
+
+Either way, you will be forwarded to a page where you will see the 
+`Run Details`. Here, you have the option to upload a `.yaml` [configurations
+file](https://docs.zenml.io/how-to/use-configuration-files) or change the 
+configuration on the go by using our editor **(powered with auto-completion and
```
Comment (schustmi) [inline docs/book/how-to/create-and-run-templates/run-a-template.md:45]: The auto completion is not really working for anyone writing normal yaml right now

File: docs/book/how-to/create-and-run-templates/run-a-template.md [hunk 2]
```diff
@@ -1,30 +1,51 @@
...
-## Run a template in code
+## Run a template using the Python SDK
+
+You can use the ZenML client to run a template:
 
 ```python
 from zenml.client import Client
 
-template = Client().get_run_template(<NAME_OR_ID>)
+template = Client().get_run_template("TEMPLATE_NAME")
```
Comment (schustmi) [inline docs/book/how-to/create-and-run-templates/run-a-template.md:23]: I'm not sure why the `<>` was removed, but isn't that at least better to show that it's a placeholder and not just a string?

File: docs/book/how-to/create-and-run-templates/run-a-template.md [hunk 3]
```diff
@@ -1,30 +1,51 @@
...
-[sign up here](https://cloud.zenml.io) get access.
+{% hint style="success" %}
+This is a [ZenML Pro](https://zenml.io/pro)-only feature. Please
+[sign up here](https://cloud.zenml.io) to get access.
 {% endhint %}
 
-<!-- ## Run a template from the dashboard -->
+Once your template is set up, you can **run** it and create a new pipeline run 
+that will be conducted on the same stack. Before it starts, you have a window
+to seamlessly readjust the configuration of your pipelines and steps.
```
Comment (schustmi) [inline docs/book/how-to/create-and-run-templates/run-a-template.md:14]:
```suggestion
that will be conducted on the same stack. Before it starts, you have the option
to modify the configuration of your pipeline and steps.
```

File: docs/book/how-to/create-and-run-templates/README.md [hunk 4]
```diff
@@ -1,15 +1,42 @@
 ---
-description: Creating and running templates
+description: Streamline Your Pipeline Execution
 ---
 
-{% hint style="warning" %}
-This is a [ZenML Pro](https://zenml.io/pro) only feature. Please 
-[sign up here](https://cloud.zenml.io) get access.
+Are you tired of constantly tweaking code to adjust your pipeline
```
Comment (schustmi) [inline docs/book/how-to/create-and-run-templates/README.md:5]: 😂😂😂

IMO this has no place in the docs, there is not a single piece of useful information in this paragraph

File: docs/book/how-to/create-and-run-templates/README.md [hunk 5]
```diff
@@ -1,15 +1,42 @@
 ---
-description: Creating and running templates
+description: Streamline Your Pipeline Execution
```
Comment (schustmi) [inline docs/book/how-to/create-and-run-templates/README.md:2]: This has 0 meaning

**Review by schustmi — CHANGES_REQUESTED** [review]

File: docs/book/how-to/create-and-run-templates/README.md [hunk 6]
```diff
@@ -1,15 +1,42 @@
...
+
+**Run Templates** are pre-defined, parameterized configurations for your ZenML 
+pipelines that can be easily executed from the ZenML dashboard or via our 
+Client/REST API. Think of them as blueprints for your pipeline runs, ready 
+to be customized on the fly.
+
+## Why Use Run Templates?
+
+- **Efficient Iteration**: Test different configurations rapidly without 
+modifying your codebase.
```
Comment (schustmi) [inline docs/book/how-to/create-and-run-templates/README.md:19]: I can do the same thing by running from my client, I can just rerun the same file without modifying my code base.

File: docs/book/how-to/create-and-run-templates/run-a-template.md (hunk 2 above)
Comment (bcdurak) [reply docs/book/how-to/create-and-run-templates/run-a-template.md:23]: This new format has no Pythonic errors, so I can use my auto-formatting to fix the format of the code even in the docs. Having any `<>`s in the code messes it up completely and it formats in a very weird way. That's why I usually use `""` and capital letters as a convenience. I can change it back though if you want.

File: docs/book/how-to/create-and-run-templates/README.md (hunk 4 above)
Comment (bcdurak) [reply docs/book/how-to/create-and-run-templates/README.md:5]: I have to say that I disagree with this. I also don't like this when phrases like these are overused, but this is simply two sentences at the start of an intro/cover page to introduce the motivation for the topic. I think it is pretty harmless here.

File: docs/book/how-to/create-and-run-templates/run-a-template.md (hunk 2 above)
Comment (schustmi) [reply docs/book/how-to/create-and-run-templates/run-a-template.md:23]: I mean there is not much to format here 😄 
Or make it "<...>", but all of that is worse IMO, because users will just copy-paste, it will run and then fail with a key error.

File: docs/book/how-to/create-and-run-templates/README.md (hunk 4 above)
Comment (schustmi) [reply docs/book/how-to/create-and-run-templates/README.md:5]: Well first of all, you don't need to tweak code to run your pipeline with a different configuration when running from a client, you modify the same yaml. So the whole thing is wrong.

For me personally, once I start reading this over-the-top hyped AI-generated mess, I stop reading because I know not to waste my time. I'll approve the PR now, I don't think we can agree on anything regarding docs tbh.

**Review by schustmi — APPROVED** [review]

File: docs/book/how-to/create-and-run-templates/run-a-template.md (hunk 1 above)
Comment (bcdurak) [reply docs/book/how-to/create-and-run-templates/run-a-template.md:45]: Not sure if I get it correctly what you mean by `normal yaml`. How can I rephrase this to fit the current state of the implementation?

File: docs/book/how-to/trigger-pipelines/use-templates-python.md [hunk 7]
```diff
@@ -0,0 +1,104 @@
...
+    Client().trigger_pipeline("training_pipeline", run_configuration=run_config)
+
+
+@pipeline
+def loads_data_and_triggers_training():
+    df = load_data()
+    trigger_pipeline(df)  # Will trigger the other pipeline
+```
+
+Read more about the [PipelineRunConfiguration](https://sdkdocs.zenml.io/latest/core_code_docs/core-config/#zenml.config.pipeline_run_configuration.PipelineRunConfiguration) and [`trigger_pipeline`](https://sdkdocs.zenml.io/0.60.0/core_code_docs/core-client/#zenml.client.Client) function object in the [SDK Docs](https://sdkdocs.zenml.io/).
```
Comment (htahir1) [inline docs/book/how-to/trigger-pipelines/use-templates-python.md:98]:
```suggestion
Read more about the [PipelineRunConfiguration](https://sdkdocs.zenml.io/latest/core_code_docs/core-config/#zenml.config.pipeline_run_configuration.PipelineRunConfiguration) and [`trigger_pipeline`](https://sdkdocs.zenml.io/latest/core_code_docs/core-client/#zenml.client.Client) function object in the [SDK Docs](https://sdkdocs.zenml.io/).
```

**Review by htahir1 — CHANGES_REQUESTED** [review]
Really small comment, just apply that and merge: -)

**Review by htahir1 — APPROVED** [review]
